Presentation.Web/Infrastructure/Middleware/CSRFProtectionMiddleware.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using System.Web.Helpers;
+using System.Web.Mvc;
+using Core.ApplicationServices.Authentication;
+using Microsoft.Owin;
+using Ninject;
+using Presentation.Web.Extensions;
+
+namespace Presentation.Web.Infrastructure.Middleware
+{
+    public class CSRFProtectionMiddleware : OwinMiddleware
+    {
+        public CSRFProtectionMiddleware(OwinMiddleware next) : base(next)
+        {
+        }
+
+        private const string XsrfHeader = "X-XSRF-TOKEN";
+        private const string XsrfCookie = "XSRF-TOKEN";
+
+        public override async Task Invoke(IOwinContext context)
+        {
+            var kernel = context.GetNinjectKernel();
+            var authenticationContext = kernel.Get<IAuthenticationContext>();
+            if (authenticationContext.Method == AuthenticationMethod.KitosToken)
+            {
+                await Next.Invoke(context);
+                return;
+            }
+
+            if (context.Request.Method.IsGet())
```

Comment:
har vi ikke en allerede der hedder IsMutation som anvendes i den der forhindrer API mutations? Den kan vi vel genbruge her for det er vel bare det omvendte